lms/djangoapps/teams/csv.py

```diff
@@ -406,21 +432,31 @@ def is_FERPA_bubble_breached(self, teamset_id, team_name):
         else:
             return True
 
-    def validate_proposed_team_size_wont_exceed_maximum(self, team_name, teamset_id):
+    def validate_team_sizes_not_exceeded(self):
         """
         Validates that the number of users we want to add to a team won't exceed maximum team size.
         """
-        if self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size is None:
-            max_team_size = self.course.teams_configuration.default_max_team_size
-        else:
-            max_team_size = self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size
+        for teamset_id in self.teamset_ids:
+            # Get max size for team-set
+            if self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size is None:
+                max_team_size = self.course.teams_configuration.default_max_team_size
+            else:
+                max_team_size = self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size
+
+            if max_team_size is not None:
+                # Get teams in teamset
+                team_names = [
+                    teamset_to_team[1] for teamset_to_team in self.user_count_by_team
+                    if teamset_to_team[0] == teamset_id
+                ]
+
+                for team_name in team_names:
+                    # Calculate team size
+                    key = (teamset_id, team_name)
+                    if (self.user_count_by_team[key] - self.user_to_remove_by_team[key]) > max_team_size:
+                        self.add_error_and_check_if_max_exceeded('Team ' + team_name + ' is full.')
```

Comment:
Maybe we can have a more useful message, like "Team <name> has a max team size of <max size> which would be exceeded by the changes" or something like that? Or even without the actual numbers, just something like, "Team X would exceed maximum team size"

Reply:
I'm just thinking that the following use case would be confusing:
- you have a team that's full
- you're trying to move some people out and some in
- you accidentally are trying to put more people in than you're taking out
- you get the error message "Team X is full"
- "Yes, i know it's full, can I not move people around if it's full??"

Reply:
Updated